core/src/main/java/org/elasticsearch/common/lucene/search/function/FieldValueFactorFunction.java

```diff
@@ -116,6 +116,11 @@ protected boolean doEquals(ScoreFunction other) {
                 Objects.equals(this.modifier, fieldValueFactorFunction.modifier);
     }
 
+    @Override
+    protected int doHashCode() {
+        return Objects.hash(boostFactor, field, modifier);
```

Comment:
I think here we are safe - these not having a hashcode method at all wasn't breaking anything other than being a trap. So it isn't being called frequently. Now its less of a trap because the hash code is more likely to be right. It could still be a performance issue, certainly.

I'd normally write the hash codes by hand but it felt more in line with the rest of the code around it to use the Objects method here.